### ISSUE TITLE
[User Story][v4][WhatsApp] Socket Write error

### ISSUE THREAD

**Issue by gargsms** [loqui/im#643]
The connections are made and the auth is duly finished, but the account never shows up.

The console logs:
"Can not write. Socket state: closed ~ BinaryWriter.prototype.flushBuffer@app://5452d126-c653-4edc-b2e4-429542c6cde3/scripts/mozillahispano/coseme.js:5012:9 [BinaryWriter]" and
"Error writing! socket-non-ready ~ self._onErrorSendDisconnected@app://5452d126-c653-4edc-b2e4-429542c6cde3/scripts/mozillahispano/coseme.js:7580:9 [ConnectionManager]"

Tried on both FxOS v2.0 emulator and Flame.

All permissions except Geolocation are set to 'Grant'.

**Comment by TitanNano** [comment]
Could you just try to connect again? 
It's working perfectly fine for me and this looks like WhatsApp is closing the connection. I experienced something similar a while ago and it went back to normal after some hours....

**Comment by ghost** [comment]
@gargsms does that affect only Loqui or are other app also affected? If yes maybe a FirefoxOS restart could fix that issue ..

**Comment by gargsms** [comment]
So I flashed my Flame with the latest FxOS nightly release today. Tried to use WhatsApp on Loqui but couldn't.

The registration SMS arrives without any problems. Then the validation is also done properly. But after the 'Synchronizing' notification appears, the screen is not taken to the 'main' section like it is with every other account (Facebook, XMPP etc.) The screen just remains on the verification section. 

I have attached a video here: https://mega.co.nz/#!y4tzwJzS!50qnL5NgcXIYWdYZXnNAVWkkO5Lo2owtgeet8ILhx5I

Sadly enough, my WebIDE is not logging anything. And I have no idea why.
I also turned Verbose App Permissions on, and this is the screenshot. Everything except Geoloc is on Grant.
![2015-02-22-19-37-31](https://cloud.githubusercontent.com/assets/5238955/6318808/7de940ce-baca-11e4-88aa-4f4dc5eb3671.png)

**Comment by TitanNano** [comment]
yes, the missing transition to the main section is connected to the not working connection to whatsapp. I can't tell whats the problem right now. Did you reinstall the Application or just update it? I can connect to WhatsApp with the latest master, so it's not a general problem

**Comment by gargsms** [comment]
I tried both the Marketplace and the latest master build.

I can, however, see the connection made to WhatsApp because the app keeps on sending pong to the WhatsApp servers. Also, I need to reverify on an Android test device after I have registered with Loqui.

For a time or two, I got CORS errors in the WebIDE. But now I don't.

When the app resumes after screen unlock, I could see "Accounts.current is not defined" in the console.

**Comment by TitanNano** [comment]
> I can, however, see the connection made to WhatsApp because the app keeps on sending pong to the  WhatsApp servers. Also, I need to reverify on an Android test device after I have registered with Loqui.

thats related to the fact that Loqui tried to establish a connection, but even when the tcp stream got disconnected as you described, Loqui keeps trying to send the pong. 

> For a time or two, I got CORS errors in the WebIDE. But now I don't.

CORS erros can only occur on connections to bosh.loqui.im. 

> When the app resumes after screen unlock, I could see "Accounts.current is not defined" in the console.

I can just guess but I assume that this is related to the fact that the whatsapp account couldn't get setup properly.

**Comment by TitanNano** [comment]
Do you get any other errors before the write error? make sure you enabled the debug mode for advanced logging.

**Comment by gargsms** [comment]
I know this will sound funny, but this is what happened exactly, and I did not get any chance to log anything.

I have always hated OpenWapp, partly because I can't close it once I start it and partly because it is not as awesome as Loqui IM. But then I could not use WhatsApp on Loqui.
So I installed OpenWapp, started it, verified my number, didn't like it a wee bit.
Then I downloaded a fresh copy of Loqui from Marketplace. With OpenWapp running in background (because I couldn't exit it) I verified my number on Loqui, and bang, my account got set up properly. No sticking up at the Synchronizing log.

Then I uninstalled OpenWapp. :stuck_out_tongue:  Now I can rest in peace, and implement things I have waited so long for in Loqui. :grinning:
